### PR TITLE
SNAP-894 Search by Approximate Age

### PR DESCRIPTION
**Pull request by ggomez-cambriasolutions**
### Jira Story

https://osi-cwds.atlassian.net/browse/SNAP-894

## Description
This PR will implement the search by approximate age feature in the Search UI.

As a user, I need to be able to enter an approximate age for a client and still get search results that will include the person I am looking for.

## Tests
- [x] I have included unit tests 
- [ ] I have included feature tests 
- [ ] I have included other tests 
- [ ] I have NOT included tests 

## Types of changes

- [x] New feature (non-breaking change which adds functionality)
- [x] Refactoring (No behavioral changes)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [x] My code follows the code style of this project.
- [x] I have ran all tests for the project.
- [x] I have ran lint/rubocop check for the changed/new files.
- [x] My code is in a stable state ready to be deployable, but not necessarily complete.
- [x] I promise on my honor as a CWDS developer to ensure I don't break the build, to check Lint/Rubocop, use best practices, to leave the code in better shape than I found it, and to have a good day.

### CODE REVIEW

File: app/javascript/common/search/age/AgeNumberSelect.jsx [hunk 1]
```diff
@@ -2,26 +2,54 @@ import React from 'react'
...
+      'searchApproximateAge',
+      number >= range.min && number <= range.max ? value : ''
+    )
+  }
+
+  render() {
+    const {id, gridClassName, value, range, disabled} = this.props
+    const options = []
+
+    for (let x = range.min; x <= range.max; x++) {
```
Comment (TylerOSI) [inline app/javascript/common/search/age/AgeNumberSelect.jsx]: I don't like using for loops, but after researching ranges a bit, it seems like this is the simplest way to accomplish this.

File: app/javascript/common/search/age/AgeNumberSelect.jsx [hunk 2]
```diff
@@ -2,26 +2,54 @@ import React from 'react'
...
-      <option key="" value="" />
-    </SelectField>
-  )
+class AgeNumberSelect extends React.Component {
+  handleClick() {
+    const {onChange} = this.props
+    onChange('searchByAgeMethod', 'approximate')
+  }
+
+  onChange({target: {value}}) {
```
Comment (TylerOSI) [inline app/javascript/common/search/age/AgeNumberSelect.jsx]: is this method ever called? It gets a little confusing when there is a method named `onChange` with a variable called `onChange`. Which one is called and why? It seems like an ambiguity problem to me.

File: spec/javascripts/components/common/search/age/AgeNumberSelectSpec.jsx [hunk 3]
```diff
@@ -7,38 +7,95 @@ const render = ({
...
+      const selectField = component.find('SelectField')
       expect(selectField.exists()).toEqual(true)
-      expect(selectField.find('option[value=""]').exists()).toEqual(true)
+      expect(selectField.find('option[value=0]').text()).toEqual('0')
+    })
+
+    it('renders a div with props', () => {
+      const component = render({})
+      expect(component.find('div').exists()).toBe(true)
+      expect(typeof component.find('div').props().onClick).toEqual('function')
```
Comment (TylerOSI) [inline spec/javascripts/components/common/search/age/AgeNumberSelectSpec.jsx]: this seems like a very useless expectation. onClick is always meant to be set to a function.

File: app/javascript/common/search/age/AgeNumberSelect.jsx (hunk 2 above)
Comment (ggomez-cambriasolutions) [reply app/javascript/common/search/age/AgeNumberSelect.jsx]: Yeah, I hear you. It's a pattern that is wide-spread across the Intake app. The onChange prop is called inside the class method onChange when the selection changes.

Comment (ggomez-cambriasolutions) [reply app/javascript/common/search/age/AgeNumberSelect.jsx]: I'll rename those method names to be more clear. Better now than never.

File: spec/javascripts/components/common/search/age/DateOfBirthDateFieldSpec.jsx [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+      const component = render({})
+      const dateFieldParent = component.find('div')
+      expect(dateFieldParent.exists()).toBe(true)
+      expect(typeof dateFieldParent.props().onClick).toEqual('function')
+      expect(dateFieldParent.props().role).toEqual('presentation')
+    })
+
+    it('renders a DateField', () => {
+      const component = render({value: '2019-03-01'})
+      console.log(component.debug())
```
[the remark this answers is not in the document]
Comment (ggomez-cambriasolutions) [reply spec/javascripts/components/common/search/age/DateOfBirthDateFieldSpec.jsx]: hmm, I wonder why the linter didn't pick this up. ok, will do.

**Pull request merged**